### DISCUSSION TITLE
RomWBW v3.2.0

### DISCUSSION THREAD

**Discussion by wwarthen** [wwarthen/RomWBW#334] [General]
Please review the [RomWBW User Guide](https://github.com/wwarthen/RomWBW/blob/master/Doc/RomWBW%20User%20Guide.pdf) document for installation and usage.

A (somewhat) complete list of changes can be found in the [ChangeLog document](https://github.com/wwarthen/RomWBW/blob/master/Doc/ChangeLog.txt).

<hr /><em>This discussion was created from the release <a href='https://github.com/wwarthen/RomWBW/releases/tag/v3.2.0'>RomWBW v3.2.0</a>.</em>

**Comment by XRBR** [comment]
Hello.
First of all, thank you very much for his work.

I don't have in-depth knowledge of microprocessors and I found this ZETA SBC project by chance, I was interested in the project because of the nostalgia for the 80s.
I arrived at your site by the author himself who already informs the direction to his site.

I was using version 3.0.1 without noticing problems. When updating to version 3.2, I noticed that it does not correctly inform the clock to the CPU and does not seem to identify the CTC. About the FDC, it detected that I don't have it.
I have attached 2 images for comparison.
My English is very bad, I didn't get to read your documentation if these situations are known bugs... Also, for errors of concordance in sentences, please forgive google.translator.

Again, thanks for sharing your work.

![romwbw_3 0 1](https://user-images.githubusercontent.com/23155655/227081334-1cb11b54-9d29-4206-b76b-0965be503c93.jpg)
![romwbw_3 2 0](https://user-images.githubusercontent.com/23155655/227081338-3bf5573d-b368-4700-af80-cde273dbaf3b.jpg)

And I just noticed that the RAM DISK capacity is also wrong. It is reporting 256K

**Comment by wwarthen** [comment]
Thank you @XRBR.  You were very clear in your description of the issues.  I appreciate the photos -- very helpful.

The FD issue was discovered by another user.  That fix will be available shortly.

The CTC and CPU speed issues are odd.  I am going to see if I can duplicate any of this on my Zeta 2.  I will post again in the next day.

Thanks,

Wayne

**Comment by wwarthen** [comment]
I found the CPU speed issue.  Stupid typo.  I guess I didn't catch this in testing because most of my systems are running at the default CPU speed, so I never noticed a failure in the dynamic speed detection.

I have not been able to replicate the CTC problem on any of my systems.  I suspect that your specific CTC chip is not quite fast enough to handle the rapid I/O port read/write sequence of the detection algorithm.  I am going to play with this some more.

Thanks,

Wayne

**Comment by XRBR** [comment]
"I found the CPU speed issue. Stupid typo."
Who never made a mistake in typing??? hahahaha

I appreciate your answer.
About the CTC, theoretically it operates at 4Mhz, but as it was bought on ebay, the serigraphy could have been faked...

I add an image of my board, where the ICs are visible.

![SBCV2](https://user-images.githubusercontent.com/23155655/227295548-b0246782-e7b0-48a3-98ff-496ac45c0ced.jpg)

**Reply by wwarthen** [reply]
Hi @XRBR,

I just pushed commit 98a33b8 which includes:

- Dynamic CPU speed detection fix
- New CTC chip detection algorithm

A new development snapshot with this commit has been posted on the  [Releases ](https://github.com/wwarthen/RomWBW/releases) page as v3.2.1-dev.2.

The floppy disk configuration issue for Zeta 2 is included as well (from a prior commit).

I just noticed your comment about the size of the RAM disk.  Actually, in v3.2, the correct size is 256K.  This is due to more RAM banks being used for CP/M 3 buffers.

I really don't know if this update will fix your CTC chip detection issue.  I was never able to recreate the problem, but I was not happy with the previous detection algorithm.  I think this one is better and hope that it works for you.

If you have time, I would appreciate it if you could try this new build. I really appreciate your problem report.

Thanks,

Wayne

**Comment by XRBR** [comment]
I just tested.
With the new release the CPU frequency normalized.
For my card, unfortunately the CTC detection remains inoperative.
Maybe it's because the CPU runs at 10Mhz and I'm using a 12,352Mhz crystal? Can this "overclock" cause the error?
Could it just be a fake message?
What is the way to detect if the CTC is really inoperative?

Sorry if I write nonsense, as I said, I don't have the knowledge to troubleshoot correctly.

What intrigues me is that version 3.0.1 does not have this CTC detection error.

Thanks.
![3 2 1](https://user-images.githubusercontent.com/23155655/227410415-e52cde30-fe1e-4f2c-9676-dc51353422e2.jpg)

**Comment by wwarthen** [comment]
Hi @XRBR,

So, first of all, it is actually possible that the CTC chip has been inoperative from the start.  v3.0.x of RomWBW did not actually test to see if the CTC exists.  It just assumed it was there.  Additionally, the Zeta would run perfectly without a working CTC.  The only function that would show a problem is the `TIMER` command which displays the system timer.  If the CTC is not working under v3.0.x, the `TIMER` command would just show zeroes.

At this point I do believe that you have a hardware issue with the CTC.  You have an NMOS variant of the chip, so it might be more sensitive to the CPU clock speed.  Alternatively, it could just be a broken chip altogether.

You can manually check the chip operation using the RomWBW monitor.  Basically, you just write 0x17 then 0xFF to the CTC port at 0x20.  Then read from the CTC port at 0x20 and the values you read should keep changing because the counter will be running.  Here is a sample session from my Zeta:

```
Monitor Ready (? for Help)
8E>O 20 17
8E>O 20 FF
8E>I 20
E0
8E>I 20
0C
8E>I 20
99
```

Assuming this fails, you could try to use a slower oscillator for your Zeta CPU clock.  Ideally, something 4 MHz or below.  If the CTC does work at the slower clock speed, then it is simply a speed issue with the chip.  Otherwise, the chip is just bad.

If you do wind up replacing the CTC chip, I suggest looking for a CMOS version.  They are available rated up to 10 MHz.  The 10 MHz CMOS chip is known to run fine at 20 MHz or even higher.

Thanks,

Wayne

**Comment by XRBR** [comment]
Hello!

Wayne, thank you very much for the information, you were right, the problem was the frequency above the supported.

I found an 8.192Mhz crystal on a video capture card and with that crystal it worked perfectly!

I ordered a Z84C3010PEC, in 1 or 2 months it should arrive and I will test it again.

Thank you very much.
![zeta_8mhz](https://user-images.githubusercontent.com/23155655/227716853-5b930d64-0755-43db-85f3-783513f6f1a1.jpg)

**Reply by wwarthen** [reply]
Excellent.  Thank you for confirming this!

-Wayne
